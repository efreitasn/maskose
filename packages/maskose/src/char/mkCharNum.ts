import { MaskCharPredicateFn } from '.';

export const MASK_CHAR_NUM_TYPE = 'MASK_CHAR_NUM_TYPE';

export type MaskCharNumObj = {
  type: typeof MASK_CHAR_NUM_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A character in the range 0 to 9 expected to be in the value to be masked
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharNum(predicateFn?: MaskCharPredicateFn): MaskCharNumObj {
  return {
    type: MASK_CHAR_NUM_TYPE,
    predicateFn
  };
}