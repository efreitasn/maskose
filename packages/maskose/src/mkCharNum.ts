const MASK_CHAR_NUM_TYPE = 'MASK_CHAR_NUM_TYPE';

export type MaskCharPredicateFn = (arg: {
  value: string;
}) => boolean;

export type MkCharNumObj = {
  type: typeof MASK_CHAR_NUM_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

export default function mkCharNum(predicateFn?: MaskCharPredicateFn): MkCharNumObj {
  return {
    type: MASK_CHAR_NUM_TYPE,
    predicateFn
  };
}