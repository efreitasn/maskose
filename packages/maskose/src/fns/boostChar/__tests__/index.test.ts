import mkBoostChar from '..';
import mkCharNum from '../../../mask/chars/num';
import mkCharBoostRepeat from '../../../boosts/char/repeat';
import mkCharBoostValueLengthEqualTo from '../../../boosts/char/valueLengthEqualTo';

it('should match the object returned by mkBoostChar()()', () => {
  const result = mkBoostChar(mkCharNum())([
    mkCharBoostRepeat(4),
    mkCharBoostValueLengthEqualTo({
      masked: 20,
      unmasked: 10
    })
  ]);

  expect(result).toMatchSnapshot();
});