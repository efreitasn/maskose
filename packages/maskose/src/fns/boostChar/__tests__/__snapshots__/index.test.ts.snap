// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the object returned by mkBoostChar()() 1`] = `
Object {
  "maskedValueLengthConditions": Array [
    Object {
      "num": 20,
      "type": "eq",
    },
  ],
  "regExp": /\\[0-9\\]/,
  "repetitions": 4,
  "type": "MASKOSE_CHAR_NUM_TYPE",
  "valueToBeMaskedLengthConditions": Array [
    Object {
      "num": 10,
      "type": "eq",
    },
  ],
}
`;
